code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -129,20 +129,23 @@
  *
  * Automatically creates more reality smashes
  */
-/datum/reality_smash_tracker/proc/Generate()
-	targets++
-	var/number = max(targets * (4-(targets-1)) - smashes,1)
+/datum/reality_smash_tracker/proc/Generate(mob/caller)
+	if(!istype(caller))
+		stack_trace("Caller is of not the right type!")
+	targets += caller
+	var/number = max(targets.len * (4-(targets.len-1)) - smashes.len,1)
 
-	for(var/i in 0 to number)
+	for(var/obj/effect/reality_smash/smash in smashes)
+		smash.alt_appearance.add_to_hud(caller)
```

Comment:
```suggestion
	for(var/iter_smash in smashes)
	    var/obj/effect/reality_smash/smash = iter_smash
		smash.alt_appearance.add_to_hud(caller)
```
Iterating through the list in this way (having iter_smash and then smash separately) is how Rohesie says to iterate through lists like this, since it will catch any null entries in the list and cause a runtime if one exists, rather than silently skip over the null.

Reply:
alright, I will also cleanup on my end and we will see how it works